key-manager-oapi.yaml

```diff
@@ -0,0 +1,227 @@
+
+        DELETE should never return a 404 response, even if all pubkeys from request.pubkeys have no extant keystores
+        nor slashing protection data.
+      security:
+        - cookieAuth: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                pubkeys:
+                  type: array
+                  description: List of public keys to delete.
+                  items:
+                    $ref: "#/components/types/Pubkey"
+      responses:
+        "200":
+          description: Success response
+          content:
+            application/json:
+              schema:
+                title: DeleteKeysResponse
+                type: object
+                properties:
+                  data:
+                    type: array
+                    description: Deletion status of all keys in `request.pubkeys` in the same order.
+                      type: object
+                      properties:
+                        status:
+                          type: string
+                          description: |
+                            - deleted: key was active and removed
+                            - not_active: slashing protection data returned but key was not active
+                            - not_found: key was not found to be removed, and no slashing data can be returned
+                            - error: unexpected condition meant the key could not be removed (the key was actually found, but we couldn't stop using it) - this would be a sign that making it active elsewhere would almost certainly cause you headaches / slashing conditions etc.
+                          enum:
+                            - deleted
+                            - not_active
+                            - not_found
+                            - error
+                        message:
+                          type: string
+                          description: error message if status == error
+                  slashing_protection:
+                    type: object
+                    description: |
+                      JSON representation of the slash protection data in format defined in EIP-3076: Slashing
+                      Protection Interchange Format.
+
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalError"
+
+components:
+  responses:
+    Unauthorized:
+      $ref: "./types/http.yaml#/Unauthorized"
+    Forbidden:
+      $ref: "./types/http.yaml#/Forbidden"
+    InternalError:
+      $ref: "./types/http.yaml#/InternalError"
+
+  securitySchemes:
+    cookieAuth: # arbitrary name for the security scheme; will be used in the "security" key later
+      type: apiKey
+      in: cookie
+      name: SESSION # cookie name
```

Comment:
I think the rough consensus around cookies vs tokens fell in favour of tokens (see https://github.com/ethereum/beacon-APIs/pull/151#discussion_r662211483). Prysm have implemented a JWT, and Lighthouse has a bearer token that's compatible.

We can specify token auth using the template here: https://swagger.io/docs/specification/authentication/bearer-authentication/

Reply:
yup in favor of a token for now as it keeps it more simple.